libtiledbsoma/src/utils/geometry/operators/io/write.cc

```diff
@@ -0,0 +1,178 @@
+#include "write.h"
+
+namespace tiledbsoma::geometry {
+
+size_t WKBSizeOperator::binary_size(const BasePoint& point) {
+    size_t size = 16;
+    if (point.z.has_value())
+        size += 8;
+    if (point.m.has_value())
+        size += 8;
+    return size;
+}
+
+size_t WKBSizeOperator::operator()(const Point& point) {
+    return 5 + binary_size(point);
+}
+
+size_t WKBSizeOperator::operator()(const LineString& linestring) {
+    if (linestring.points.size() == 0)
+        return 9;
+
+    return 9 +
+           linestring.points.size() * binary_size(linestring.points.front());
+}
+
+size_t WKBSizeOperator::operator()(const Polygon& polygon) {
+    size_t size = 1 + 4 + 4;
```

Comment:
Would you consider moving these values as top-level consts to improve readability? For example:

`const size_t WKB_BYTE_ORDER_SIZE = 1;`
`const size_t WKB_GEOEMTRY_TYPE_SIZE = 4;`
`const size_t WKB_SRID_SIZE = 4;`
`const size_t WKB_GEOMETRY_HEADER_SIZE = WKB_BYTE_ORDER_SIZE + WKB_GEOEMTRY_TYPE_SIZE + WKB_SRID_SIZE;`

This would make the code clearer, especially for people (like me) who may not be as familiar with the structure of WKB files.